Sources/InsulinDeliveryServiceKit/Mock/VirtualInsulinDeliveryPump.swift

```diff
@@ -0,0 +1,751 @@
+            } else if isConnected && status.isActiveBolusDeliveryInProgress() {
+                status.updateDelivery()
+            }
+            delegate?.pumpDidUpdateState(self)
+            delegate?.pumpConnectionStatusChanged(self)
+        }
+    }
+
+    var isAuthenticated: Bool {
+        get {
+            status.isAuthenticated
+        }
+        set {
+            if status.isAuthenticated != newValue {
+                status.isAuthenticated = newValue
+            }
+        }
+    }
+
+    var isAwaitingConfiguration = false
+    
+    func getTime(using timeZone: TimeZone, completion: @escaping ProcedureTimeCompletion) {
+        scheduleTask(after: schedulerDelay) {
+            completion(.success(Date()))
+        }
+    }
+
+    func setTime(_ date: Date = Date(), using timeZone: TimeZone, completion: @escaping ProcedureResultCompletion) {
+        scheduleTask(after: schedulerDelay) {
+            completion(.success)
+        }
+    }
+
+    func setOOBString(_ oobString: String) {
+        if let oobData = oobString.data(using: .utf8) {
+            state.securityManagerConfiguration.oobRandomNumber = oobData
+            prepareForNewPump()
+        }
+    }
+
+    var deviceInformation: DeviceInformation? {
+        get {
+            return state.deviceInformation
+        }
+        set {
+            if state.deviceInformation != newValue {
+                state.deviceInformation = newValue
+            }
+            if isConnected {
+                triggerStoppedAnnunciationIfNeeded()
+            }
+        }
+    }
+
+    private let schedulerDelay: TimeInterval
+
+    var currentAnnunciationIdentifier: AnnunciationIdentifier = 1
+
+    var errorOnNextComms: DeviceCommError?
+
+    private var lowReservoirDidAlert: Bool = false
+
+    var stoppedAnnunciationTimer: Timer?
+    
+    var authenticationError: DeviceCommError?
+    
+    var stoppedNotificationDelay = TimeInterval.hours(1)
+
+    var uncertainDeliveryEnabled: Bool = false {
+        didSet {
+            if !uncertainDeliveryEnabled {
+                resolveUncertainDelivery()
+            }
+        }
+    }
+
+    var uncertainDeliveryCommandReceived: Bool = false
+
+    private var pendingResponse: (() -> Void)?
+
+    init(status: MockInsulinDeliveryPumpStatus? = nil, schedulerDelay: TimeInterval = VirtualInsulinDeliveryPump.defaultSchedulerTimeDelay) {
+        self.schedulerDelay = schedulerDelay
+        guard let status = status else {
+            lockedStatus = Locked(MockInsulinDeliveryPumpStatus())
+            return
+        }
+        lockedStatus = Locked(status)
+    }
+
+    @Sendable
+    private func scheduleTask(after time: TimeInterval, task: @escaping () -> Void) {
+        DispatchQueue.global(qos: .userInitiated).asyncAfter(deadline: .now() + time) {
+            if self.isConnected && self.errorOnNextComms == nil {
+                self.lastCommsDate = Date()
+            }
+            task()
+        }
```

Comment:
nit: I think this same could be accomplished with a Task:
```suggestion
        Task(priority: . userInitiated) { [weak self] in
            try? await Task.sleep(nanoseconds: NSEC_PER_SEC * time)
            
            if self?.isConnected == true && self?.errorOnNextComms == nil {
                self?.lastCommsDate = Date()
            }
            
            task()
        }
```